Analysis/src/TypeInfer.cpp

```diff
@@ -3412,6 +3412,20 @@ TypeId TypeChecker::checkLValueBinding(const ScopePtr& scope, const AstExprIndex
                 return prop->type();
             }
         }
+        else if (get<IntersectionType>(exprType))
```

Comment:
Needs a flag.
`LUAU_FASTFLAGVARIABLE(Luau*something*, false)` at top of the file.
and
`ScopedFastFlag sff{"Luau*something*", true};`
in tests.